# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/ImageBridgeChild.h
# Commit: 204b899f436d
# Full Hash: 204b899f436d99f0eae759af3ccfcbaabfbbb428
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 09:34:54
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 3f5baab34b1f -r 204b899f436d gfx/layers/ipc/ImageBridgeChild.h
--- a/gfx/layers/ipc/ImageBridgeChild.h	Thu Jun 11 06:36:55 2020 +0000
+++ b/gfx/layers/ipc/ImageBridgeChild.h	Thu Jun 11 06:37:35 2020 +0000
@@ -15,7 +15,6 @@
 #include "mozilla/Atomics.h"
 #include "mozilla/RefPtr.h"            // for already_AddRefed
 #include "mozilla/ipc/SharedMemory.h"  // for SharedMemory, etc
-#include "mozilla/layers/CanvasClient.h"
 #include "mozilla/layers/CompositableForwarder.h"
 #include "mozilla/layers/CompositorTypes.h"
 #include "mozilla/layers/PImageBridgeChild.h"
@@ -32,7 +31,6 @@
 
 namespace layers {
 
-class AsyncCanvasRenderer;
 class ImageClient;
 class ImageContainer;
 class ImageContainerListener;
@@ -193,9 +191,6 @@
   RefPtr<ImageClient> CreateImageClientNow(CompositableType aType,
                                            ImageContainer* aImageContainer);
 
-  already_AddRefed<CanvasClient> CreateCanvasClient(
-      CanvasClient::CanvasClientType aType, TextureFlags aFlag);
-  void UpdateAsyncCanvasRenderer(AsyncCanvasRenderer* aClient);
   void UpdateImageClient(RefPtr<ImageContainer> aContainer);
 
   /**
@@ -213,22 +208,11 @@
   virtual ~ImageBridgeChild();
 
   // Helpers for dispatching.
-  already_AddRefed<CanvasClient> CreateCanvasClientNow(
-      CanvasClient::CanvasClientType aType, TextureFlags aFlags);
-  void CreateCanvasClientSync(SynchronousTask* aTask,
-                              CanvasClient::CanvasClientType aType,
-                              TextureFlags aFlags,
-                              RefPtr<CanvasClient>* const outResult);
-
   void CreateImageClientSync(SynchronousTask* aTask,
                              RefPtr<ImageClient>* result,
                              CompositableType aType,
                              ImageContainer* aImageContainer);
 
-  void UpdateAsyncCanvasRendererNow(AsyncCanvasRenderer* aClient);
-  void UpdateAsyncCanvasRendererSync(SynchronousTask* aTask,
-                                     AsyncCanvasRenderer* aWrapper);
-
   void FlushAllImagesSync(SynchronousTask* aTask, ImageClient* aClient,
                           ImageContainer* aContainer);
 